### ISSUE TITLE
An array of a custom type struct does not conform to KeyStringDecodable

### ISSUE THREAD

**Issue by abbasmousavi** [vapor/core#110]
This model 

```
struct Pet: PostgreSQLJSONType, Codable {
    var name: String
    var type: String
}

struct Person: PostgreSQLModel, Migration{
    var id: Int?
    var title: String
    var pets: [Pet]
}
```

causes this fatal error:

`Pet does not conform to KeyStringDecodable`

**Comment by 0xTim** [comment]
This will be due to Conditional Conformances half backed implementation I believe. I think you might need to manually extend `[Pet]` to `KeyStringDecodable`

**Comment by tanner0101** [comment]
@0xTim I have a couple of ideas on how to fix this. Hopefully one of them works.

**Comment by tanner0101** [comment]
Fixed by #111

**Comment by abbasmousavi** [comment]
@tanner0101 
Now the migration is Ok and there is a jsonb column in database for `pets`. but when I try to write a `Person` to database I get this error

```
[ ERROR ] PostgreSQLDiagnosticResponse.jsonb_recv: ERROR: unsupported jsonb version number 123 (EngineServer.swift:144)
[ DEBUG ] Possible causes for PostgreSQLDiagnosticResponse.jsonb_recv: unsupported jsonb version number 123 (EngineServer.swift:150)
[ ERROR ] PostgreSQLDiagnosticResponse.jsonb_recv: ERROR: unsupported jsonb version number 123 (EngineServer.swift:144)
[ DEBUG ] Possible causes for PostgreSQLDiagnosticResponse.jsonb_recv: unsupported jsonb version number 123 (EngineServer.swift:150)```
